server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -60,8 +64,21 @@ object Email {
       refinedValue => refinedValue)
   }
 
-  def from(message: (MessageId, Seq[MessageResult])): Email =
-    Email(message._1, sanitizeSize(message._2.head.getSize))
+  def from(message: (MessageId, Seq[MessageResult])): Try[Email] = {
+    val firstMessage: MessageResult = message._2.head
```

Comment:
use  headOption, otherwise it will throw if the Seq is empty.
Or as you have introduced cats, you can use 'NonEmptyList' instead of 'Seq'